### DISCUSSION TITLE
YouTube Watch → Embed

### DISCUSSION THREAD

**Discussion by mattieb** [mshibanami/redirect-web#17] [Redirect Rules]
If you, like me, have no use for the comments section, related videos, or annoying popup messages on YouTube, this redirect rule will take you from YouTube "watch" links to "embed" links automatically—the latter fills the screen.

**UPDATE:** Improved version that removes extra query parameters. Original version is at end of post.

- Redirect from `https://(www.|m.)youtube.com/watch\?v=([^&]*).*`
- Redirect to `https://www.youtube.com/embed/$2`

_Examples:_

- `https://www.youtube.com/watch?v=jfKfPfyJRdk` → `https://www.youtube.com/embed/jfKfPfyJRdk`
- `https://m.youtube.com/watch?v=jfKfPfyJRdk` → `https://www.youtube.com/embed/jfKfPfyJRdk`
- `https://www.youtube.com/watch?v=jfKfPfyJRdk&foo=bar` → `https://www.youtube.com/embed/jfKfPfyJRdk`

_Export:_

```json
{
  "appVersion" : "5.1.1",
  "bundleID" : "io.github.mshibanami.RedirectWebForSafari",
  "createdAt" : "2024-01-06T14:05:55.300Z",
  "kind" : "RedirectList",
  "redirects" : [
    {
      "captureGroupProcesses" : [

      ],
      "comments" : "",
      "destinationURLPattern" : "https:\/\/www.youtube.com\/embed\/$2",
      "exampleURLs" : [
        "https:\/\/www.youtube.com\/watch?v=jfKfPfyJRdk",
        "https:\/\/m.youtube.com\/watch?v=jfKfPfyJRdk",
        "https:\/\/www.youtube.com\/watch?v=jfKfPfyJRdk&foo=bar"
      ],
      "excludeURLPatterns" : [

      ],
      "isEnabled" : true,
      "kind" : "Redirect",
      "sourceURLPattern" : {
        "type" : "regularExpression",
        "value" : "https:\/\/(www.|m.)youtube.com\/watch\\?v=([^&]*).*"
      },
      "title" : "YouTube Watch → Embed"
    }
  ]
}
```

----

_Original version:_

- Redirect from `https://(www.|m.)youtube.com/watch\?v=(.*)` **(regular expression)**
- Redirect to `https://www.youtube.com/embed/$2`

_Examples:_

- `https://www.youtube.com/watch?v=jfKfPfyJRdk` → `https://www.youtube.com/embed/jfKfPfyJRdk`
- `https://m.youtube.com/watch?v=jfKfPfyJRdk` → `https://www.youtube.com/embed/jfKfPfyJRdk`

_Export:_

```json
{
  "appVersion" : "5.1.1",
  "bundleID" : "io.github.mshibanami.RedirectWebForSafari",
  "createdAt" : "2024-01-03T16:55:11.693Z",
  "kind" : "RedirectList",
  "redirects" : [
    {
      "captureGroupProcesses" : [

      ],
      "comments" : "",
      "destinationURLPattern" : "https:\/\/www.youtube.com\/embed\/$2",
      "exampleURLs" : [
        "https:\/\/www.youtube.com\/watch?v=jfKfPfyJRdk",
        "https:\/\/m.youtube.com\/watch?v=jfKfPfyJRdk"
      ],
      "excludeURLPatterns" : [

      ],
      "isEnabled" : true,
      "kind" : "Redirect",
      "sourceURLPattern" : {
        "type" : "regularExpression",
        "value" : "https:\/\/(www.|m.)youtube.com\/watch\\?v=(.*)"
      },
      "title" : "YouTube Watch → Embed"
    }
  ]
}
```

**Comment by mshibanami** [comment]
Hi @mattieb, this works great for me! 

I recently released a new version of Redirect Web, and it has a new "Library" section where users can download useful predefined rules. If you don't mind, could I add your rule to the library?

By the way, I think adding `?autoplay=1` would be great as it auto-plays the video like the original video URL. 😊

**Comment by mattieb** [comment]
@mshibanami By all means! Please feel free to make any changes you think make it work even better, too.

**Reply by mshibanami** [reply]
Thanks! I've added it with a thumbnail. Now it's on the library. 🙏
